src/number/format-decimal.js

```diff
@@ -19,6 +19,15 @@ module.exports = function(value, localeData, options) {
 	var ret = '';
 	var processedDecimal = (valueStr.indexOf('.') < 0);
 	var digitNum = 0;
+	var currentGroupSize ;
+	var groupSizes;
+	if ( Array.isArray( localeData.groupSize) ) {
+		groupSizes = localeData.groupSize.slice(); 	// make a copy of the groupsizes
+		currentGroupSize = groupSizes.shift();		// pop the first groupSize from the front
+	} else {
+		groupSizes = [];
+		currentGroupSize = localeData.groupSize;
+	}
```

Comment:
Instead of `slice()` to copy and `shift()` to pop, could you just move a pointer around within the existing array? @omsmith and I are worried about the unnecessary performance hits of this array manipulation if there are many numbers being rendered on the page.